File: 3132-find-the-integer-added-to-array-ii/3132-find-the-integer-added-to-array-ii.cpp
class Solution {
public:
    bool check(vector<int>& nums1, vector<int>& nums2, int k){
        int count = 0, j = 0;
        for(int i = 0; i < nums1.size() && j < nums2.size(); i++){
            if(nums1[i] + k != nums2[j]) count++;
            else j++;
        }
        if(count > 2) return 0;
        return 1;
    }
    int minimumAddedInteger(vector<int>& nums1, vector<int>& nums2) {
        sort(nums1.begin(), nums1.end());
        sort(nums2.begin(), nums2.end());
        int minEle = INT_MAX;  
        for(int i = 0; i < nums1.size(); i++){
            int x = nums2[0] - nums1[i];
            if(check(nums1, nums2, x)){
                minEle = min(minEle, x);
            }
        }
        return minEle;
    }
};

File: 3132-find-the-integer-added-to-array-ii/3132-find-the-integer-added-to-array-ii.java
class Solution {
    public int minimumAddedInteger(int[] nums1, int[] nums2) {
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        int size1=nums1.length,size2=nums2.length;
        Map<Integer,Integer> nm=new HashMap<>();
        for(int i=0;i<size1;i++)
        {
            for(int j=0;j<size2;j++)
            {
                int val=nums2[j]-nums1[i];
                nm.put(val, nm.getOrDefault(val,0)+1);
            }
        }
        int mini=Integer.MAX_VALUE;
        for(int k:nm.keySet())
        {
            int arr[]=new int[size1];
            for(int i=0;i<size1;i++)
            {
                arr[i]=nums1[i]+k;
            }
            Arrays.sort(arr);
            if(check(arr,nums2))
            {
                mini=Math.min(mini,k);
            }
        }
        return mini;
    }
    private boolean check(int nums1[], int nums2[])
    {
        int i=0,j=0,val=2;
        while(i<nums1.length && j<nums2.length)
        {
            if(nums1[i] == nums2[j])
            {
                i++;
                j++;
            }
            else
            {
                i++;
                val--;
                if(val < 0)
                {
                    return false;
                }
            }
        }
        return j == nums2.length && (i-j) <= 2;
    }
}